python/pyarrow/tests/test_dataset.py

```diff
@@ -4505,9 +4507,21 @@ def test_write_dataset_s3_put_only(s3_server):
     ).to_table()
     assert result.equals(table)
 
-    with pytest.raises(OSError, match="Access Denied"):
+    # Passing create_dir is fine if the bucket already exists
+    ds.write_dataset(
+        table, "existing-bucket", filesystem=fs,
+        format="feather", create_dir=True, partitioning=part,
+        existing_data_behavior='overwrite_or_ignore'
+    )
+    # check roundtrip
+    result = ds.dataset(
+        "existing-bucket", filesystem=fs, format="ipc", partitioning="hive"
+    ).to_table()
+    assert result.equals(table)
+
+    with pytest.raises(OSError, match="Bucket does not exist"):
         ds.write_dataset(
-            table, "existing-bucket", filesystem=fs,
+            table, "non-existing-bucket", filesystem=fs,
```

Comment:
To retain the original purpose of the test, should we use an `fs` with allow_create_bucket set to True? So that it raises the original error? (even if the user sets this parameter to True, we still need to honor the limitations / the create_dir keyword)

Reply:
Sure, though I think I want to test both cases.